lib/opencensus/stats/aggregation/distribution.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+
+module OpenCensus
+  module Stats
+    module Aggregation
+      # Distribution aggregation type
+      class Distribution
+        # Invalid bucket types.
+        class InvalidBucketsError < StandardError; end
+
+        # @return [Array<Integer>,Array<Float>] Bucket boundries.
+        attr_reader :buckets
+
+        # @private
+        # @param [Array<Integer>,Array<Float>,] buckets Buckets boundries
+        # for distribution
+        # aggregation.
+        # @raise [InvalidBucketsError] If any bucket value is nil.
+        def initialize buckets
+          if buckets.nil? || buckets.empty?
+            raise InvalidBucketsError, "buckets should not be nil or empty"
+          end
+
+          if buckets.any?(&:nil?)
+            raise InvalidBucketsError, "buckets value should not be nil"
+          end
+
+          buckets.reject! { |v| v < 0 }
```

Comment:
Better to create a new object rather than modifying the existing object in place. e.g.
```
@buckets = buckets.reject { |v| v < 0 }
```

Otherwise callers can unwittingly send the same list of buckets into two constructors and then the list gets shared between them, which can cause one object to clobber the other.